nix/ext/pg_graphql.nix

```diff
@@ -27,7 +27,9 @@ buildPgrxExtension_0_12_9 rec {
   # Setting RUSTFLAGS in env to ensure it's available for all phases
   env = lib.optionalAttrs stdenv.isDarwin {
     POSTGRES_LIB = "${postgresql}/lib";
-    PGPORT = if (lib.versions.major postgresql.version) == "17" then "5440" else "5439";
+    PGPORT = toString (5430 + 
+      (if builtins.match ".*_.*" postgresql.version != null then 1 else 0) +  # +1 for OrioleDB
+      ((builtins.fromJSON (builtins.substring 0 2 postgresql.version)) - 15) * 2);  # +2 for each major version
```

Comment:
Hm, this is confusing. Why does pg_graphql needs to know anything about the PGPORT?

Reply:
@steve-chavez its part of the pgrx build process and so needs to be handled first every pgrx extension at build time